NAAM.md

```diff
@@ -0,0 +1,278 @@
+advertisement specified by IPNI. Similar to a regular advertisement, an IPNS record may be removed
+by publishing an advertisement that:
+
+* links to the previous advertisement in `PrevousID` link as specified by IPNI,
+* uses the same `Provider` and `ContextID` values as before,
+* sets `IsRm` to `true`, and
+* sets `Entries` link to `NoEntries` as specified by IPNI.
+
+Once ingested, indexer nodes will eventually remove the Naam advertisement metadata along with the
+publisher information.
+
+### Resolving IPNS Records
+
+IPNS record resolution involves finding the IPNS record associated to a given IPNS key, i.e.
+`/ipns/<peer-id>`. As eluded to earlier, Naam uses the find API already provided by the network
+indexers to achieve this. To resolve an IPNS record, a resolver:
+
+* calculates the multihash of the given IPNS key as its SHA-256,
+* looks up the provider info for that multihash via the IPNI find API,
+* decodes the metadata from the provider info as IPNS record,
+* validates the record, and
+* if valid returns its value.
+
+## Limitations
+
+### IPNS Record Size
+
+The maximum size of IPNS records supported by Naam is 1 KiB, even though IPNS specification suggests
+a maximum limit of 10 KiB. The 1 KiB limit currently imposed by network indexers on the maximum size
+of advertisement metadata.
+
+Basic tests show that a 1KiB limit is sufficiently large for storing simple IPNS records pointing to
+IPFS paths, specially for records that do not need the peer's public key to be embedded into IPNS
+records. For example, a Ed25519 peer ID also captures the peer's public key. A marshalled IPNS
+record with such key type and empty value constructed
+using [`go-ipns`](https://github.com/ipfs/go-ipns) library comes to `257` bytes, which leaves
+sufficient space for encoding fairly long IPFS paths as the record value.
+
+### Record Validity Duration
+
+The indexers will maintain information from inactive providers for up to a week. This means IPNS
+records published by Naam with EOL of beyond a week should be re-published to remain resolvable.
+
+## Security
+
+### Advertisement Verifiability
+
+Naam advertisements will benefit from all the verifiability features of regular advertisements. They
+are signed by the IPNS record publisher and are verified for validity before being ingested by the
+network indexers.
+
+### Reputation
+
+Similar to the IPNI indexing protocol, the use of immutable advertisement chain by Naam provides a
+verifiable and immutable catalog of IPNS records exposed by a network participant. Access to such
+historical data enables verifiable analysis of participant behaviour, and offers an opportunity to
+design and develop reputation systems, which ultimately can contribute to the health of the network.
+As far as I know, this is something that is not currently offered by the existing implementations of
+IPNS.
+
+Even though, the specification of such reputation system is beyond the scope of this document, it is
```

Comment:
Is the idea that you could check the stability of advertisements?

Reply:
The idea is to check that an IPNS key associated to a peer can satisfy the path it has published in its IPNS record. This is similar to retrieval reputation calculation based on advertisement chain: Can a peer provide the data it advertises.

Because advertisement chain is signed by the peer we can apply the same validation to IPNS records.